### ISSUE TITLE
Ports called asynchronously if inside Cmd.batch

### ISSUE THREAD

**Issue by lazamar** [elm/core#989]
# tl;dr
- Elm 0.19 allows catching events and sending them through ports in the same tick of the event loop.
- If we use `Cmd.batch` the port call does not happen in the same tick of the event loop. 

# Problem Statement
Certain DOM events need to be handled synchronously. Elm now allows us to do that through a port. However, if when dispatching the event to a port we use `Cmd.batch` and send other events with it, this port call is now handled asynchronously making the handling of certain events not possible again.

Here are the return values I tested for the `update` function and their outcome.
- `(model, sendToPort event)` - gets called synchronously as expected
- `(model, Cmd.batch [ sendToPort event]  )` - gets called synchronously as expected
- `(model, Cmd.batch [ sendToPort event, otherCmd ])` - gets called asynchronously, unlike expected.

If we use `Cmd.batch` with a list containing only the port command, the call is still synchronous. This means that the amount of elements in the batch list changes the behaviour of the program.

# SSCE

In this example I'm using `event.dataTransfer.setData`, which must be called synchronously in the callback for the `dragstart` event. 
- If called synchronously it adds the data specified to `event.dataTransfer.items`.
- If called asynchronously the invocation of `setData` is silently ignored. 

![peek 2018-09-18 17-50](https://user-images.githubusercontent.com/11493705/45703367-06d0a800-bb6c-11e8-8894-c4c15690e98e.gif)

``` elm
port withBatch : Decode.Value -> Cmd msg
port withoutBatch : Decode.Value -> Cmd msg

type Msg = Noop  | WithoutBatch Decode.Value | WithBatch Decode.Value

update msg model =
    case msg of
        WithoutBatch event -> ( model, withoutBatch event )
        WithBatch event ->      ( model, Cmd.batch [ withBatch event, anotherCmd])
        Noop -> ( Model, Cmd.none )

anotherCmd = Task.perform identity (Task.succeed Noop)
```

The JS part.
``` javascript
	    var app = Elm.Main.init({ node: document.querySelector('main') })
	    
	    function addItemToEvent(event) {
	        console.log("Before adding item:", event.dataTransfer.items.length);
	        event.dataTransfer.setData("elm", "rocks");
	        console.log("After adding item:", event.dataTransfer.items.length);
	    }
	    
	    app.ports.withoutBatch.subscribe(addItemToEvent);
	    app.ports.withBatch.subscribe(addItemToEvent);
```

The full code is here https://ellie-app.com/3nGpxLkr7Zga1, however, for some reason Ellie is not handling the synchronous port call correctly. I recorded the gif running the code in Ellie in Elm 0.19.0 in my computer.

**Comment by lazamar** [comment]
I tracked it down to a more specific issue.
If I dispatch ` Task.perform identity (Task.succeed MyMsg)` together with a port, the port call becomes asynchronous.

Here is a demo that works in Ellie showing that removing the extra command makes port call synchronous again.
https://ellie-app.com/3nJ3z59nCdQa1

![peek 2018-09-18 19-10](https://user-images.githubusercontent.com/11493705/45707392-9f6c2580-bb76-11e8-81a2-58b9fa4894e7.gif)
